lib/Sapi.php

```diff
@@ -113,6 +113,12 @@ public static function sendResponse(ResponseInterface $response): void
                     if ($copied <= 0) {
                         break;
                     }
+                    // Abort on client disconnect.
+                    // With ignore_user_abort(true), the script is not aborted on client disconnect.
+                    // To avoid reading the entire stream and dismissing the data afterward, check between the chunks if the client is still there.
+                    if (1 === ignore_user_abort() && 1 === connection_aborted()) {
```

Comment:
I'd assume 0 means false and 1 is true?
If so, this code falsely breaks when the connection was aborted AND we want to ignore that?

Or did i mix something up?

Reply:
https://www.php.net/manual/en/function.ignore-user-abort.php

`1 === ignore_user_abort()` means "ignore user abort has been enabled" (somewhere in "the system" or "the code" prior to this point)

`1 === connection_aborted()` means that a connection has "gone away".

When these 2 things both happen, we are deciding, for this "manual" stream-copy loop, we do want to stop feeding the stream copy (even though "the system" has enabed `ignore_user_abort()`). Because we know that it is pointless to keep looping and pushing out chunk-size bytes each time.